jdk/test/java/lang/VirtualThread/VirtualThreadPinnedEventThrows.java

```diff
@@ -0,0 +1,111 @@
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Test parking when pinned and emitting the JFR VirtualThreadPinnedEvent throws
+ * @run junit VirtualThreadPinnedEventThrows
+ */
+
+// @modules java.base/jdk.internal.event
+// @compile/module=java.base jdk/internal/event/VirtualThreadPinnedEvent.java
+import java.lang.ref.Reference;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.LockSupport;
+//import jdk.internal.event.VirtualThreadPinnedEvent;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class VirtualThreadPinnedEventThrows {
+
+    /**
+     * Test parking when pinned and creating the VirtualThreadPinnedEvent fails with OOME.
+     */
+    @Test
+    void testVirtualThreadPinnedEventCreateThrows() throws Exception {
+        VirtualThreadPinnedEvent.setCreateThrows(true);
+        try {
+            testParkWhenPinned();
+        } finally {
+            VirtualThreadPinnedEvent.setCreateThrows(false);
+        }
+    }
+
+    /**
+     * Test parking when pinned and VirtualThreadPinnedEvent.begin fails with OOME.
+     */
+    @Test
+    void testVirtualThreadPinnedEventBeginThrows() throws Exception {
+        VirtualThreadPinnedEvent.setBeginThrows(true);
+        try {
+            testParkWhenPinned();
+        } finally {
+            VirtualThreadPinnedEvent.setBeginThrows(false);
+        }
+    }
+
+    /**
+     * Test parking when pinned and VirtualThreadPinnedEvent.commit fails with OOME.
+     */
+    @Test
+    void testVirtualThreadPinnedEventCommitThrows() throws Exception {
+        VirtualThreadPinnedEvent.setCommitThrows(true);
+        try {
+            testParkWhenPinned();
+        } finally {
+            VirtualThreadPinnedEvent.setCommitThrows(false);
+        }
+    }
+
+    /**
+     * Test parking a virtual thread when pinned.
+     */
+    private void testParkWhenPinned() throws Exception {
+        Object lock = new Object();
+        try {
+            // var completed = new AtomicBoolean();
+            AtomicBoolean completed = new AtomicBoolean();
+            Thread thread = Thread.startVirtualThread(() -> {
+                synchronized (lock) {
+                    LockSupport.park();
+                    completed.set(true);
+                }
+            });
+
+            // wait for thread to park
+            Thread.State state;
+            while ((state = thread.getState()) != Thread.State.WAITING) {
+                assertTrue(state != Thread.State.TERMINATED);
+                Thread.sleep(10);
+            }
+
+            // unpark and check that thread completed without exception
+            LockSupport.unpark(thread);
+            thread.join();
+            assertTrue(completed.get());
+        } finally {
+//            Reference.reachabilityFence(lock);
```

Comment:
Now that `Reference::reachabilityFence` is not supported by JDK 8, you may want to remove this try-finally clause.

Reply:
Sure, I have made the changes.